### PR TITLE
IndicatorView IndicatorTemplate Binding

### LINKED ISSUE #9597: [Enhancement] Make IndicatorView pass the current item for it's DataTemplate
## Summary
Right now on 4.4.0.991640 when an IndicatorView's indicator template is invoked `null` is setted to it's `DataTemplate` `BindingContext` making it impossible to do a more customized configuration, I would like to either include the ItemSource used to create that indicator or another data that would be usefull to customize and Indicator

## API Changes
No API changes on the `IndicatorView`

### Possible solution 1
IndicatorView's DataTemplate receive the ItemSource used to generate it in it's BindingContext and can be used as such:
Consider that I will be using an string[] as ItemSources
```xaml
<IndicatorView
	Margin="20"
	ItemsSource="{Binding DataSource}" <!-- string[] { 1, 2, 3, 4} -->
	Position="{Binding Position, Source={Reference car}}"> <!-- binding to my CarouselView -->
	<IndicatorView.IndicatorTemplate>
		<DataTemplate>
			<StackLayout x:Name="container">
				<Frame
					Padding="0"
					WidthRequest="10"
					HeightRequest="10"
					CornerRadius="5">
					<Label
						Text="{Binding .}" />
				</Frame>
			</StackLayout>
		</DataTemplate>
	</IndicatorView.IndicatorTemplate>
</IndicatorView>
``` 
this would enable us to make a more complex indicator's as such
image

### Possible solution 2
Better still would be if instead of automatically settings `BackgroundColors` of our top element as it's currently is would be to pass an intermediate class to represent this item's viewmodel as such:
```csharp
public class IndicatorViewModel
{
    public int Position { get; set; }
    public bool IsSelected { get; set; }
    public Color SelectedColor { get; set; } //maybe Selected Color and Color would be not necessary
    public Color Color { get; set; }
    public object Item { get; set; } //Current ItemSource
}
```
That would also enable us to do this more complex Indicator's and maybe more than before, not sure about speed and memory

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change

Set BindingContext for IndicatorTemplate view

### Issues Fixed

Fixes https://github.com/xamarin/Xamarin.Forms/issues/9597, https://github.com/xamarin/Xamarin.Forms/issues/10117

| Before | After |
|:-:|:-:|
| image | image |

### CODE REVIEW

File: src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs [hunk 1]
```diff
@@ -139,6 +141,7 @@ void AddExtraIndicatorItems()
 					HeightRequest = size,
 					CornerRadius = _indicatorView.IndicatorsShape == IndicatorShape.Circle ? (float)size / 2 : 0
 				};
+				indicator.BindingContext = items[i];
```
Comment (PureWeen) [inline src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs:144]: Can we add the indicators as logical children to the indicatorview?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs (hunk 1 above)
Comment (VladislavAntonyuk) [reply src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs:144]: do you mean call `AddLogicalChild(indicator);`?

**Comment by tranb3r** [comment]
@VladislavAntonyuk @PureWeen 
What is blocking this PR? Thanks

**Comment by mattleibow** [comment]
@VladislavAntonyuk could you rebase so we can get another CI build?

**Comment by VladislavAntonyuk** [comment]
> @VladislavAntonyuk could you rebase so we can get another CI build?

@mattleibow done

File: src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs (hunk 1 above)
Comment (PureWeen) [reply src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs:144]: Apologies on the late response. Looking closer I realize that suggestion isn't correct. 

Out of curiosity, do you think this code could get slightly reworked to just use BindableLayout?  This class feels like it's kind of just re-implementing BindableLayout. 

I worry about other issues cropping up inside the IndicatorStackLayout that have already been considered by BindableLayout

Comment (VladislavAntonyuk) [reply src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs:144]: I am not familiar with BindableLayout. I will take a look if we can replace the layout.

**Comment by tranb3r** [comment]
Is it possible to merge this PR asap? 
It's required for the solution @VladislavAntonyuk suggested months ago for a TabView control.
Thanks.

**Comment by tranb3r** [comment]
@PureWeen Is it possible to merge this PR in SR3?
It's required for the solution @VladislavAntonyuk suggested months ago for a TabView control in Maui CommunityToolkit.
Thanks.

**Review by rmarinho — DISMISSED** [review]

File: src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs (hunk 1 above)
Comment (rmarinho) [reply src/Controls/src/Core/IndicatorView/IndicatorStackLayout.cs]: Yes it could @PureWeen

**Review by jsuarezruiz — DISMISSED** [review]
There are build errors:
`D:\a\_work\1\s\src\Controls\src\Core\IndicatorView\IndicatorStackLayout.cs(159,38): error CS0103: The name '_indicatorViewPropertyChanged' does not exist in the current context [D:\a\_work\1\s\src\Controls\src\Core\Controls.Core.csproj::TargetFramework=net8.0-android]`

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
Failing test on core is unrelated

**Pull request merged**